src/base/emu-i386/simx86/cpu-emu.c

```diff
@@ -810,7 +812,7 @@ void init_emu_cpu(void)
  */
 void e_gen_sigalrm(void)
 {
-	if(!IS_EMU())
+	if(!IS_EMU() && !(CEmuStat & CeS_INSTREMU))
```

Comment:
Maybe IS_EMU() macro should be
adjusted instead? Its used somewhere,
we need to make sure all callers of it
are aware about a possible mode switch?

Reply:
I looked at all the callers of IS_EMU(), the ones in e_invalidate can use IS_EMU_JIT() now.
```
src/base/core/emu.c:    if (IS_EMU()) {  // This one is for leaving so outside instremu context 
src/base/core/int.c:    if (IS_EMU()) { // This one for ecpuoff, not important
src/base/emu-i386/simx86/cpu-emu.c:     if(!IS_EMU() && !(CEmuStat & CeS_INSTREMU)) // see above
src/base/emu-i386/simx86/cpu-emu.c:     if (IS_EMU() && iniflag) { // leave_cpu_emu, called from emu.c (first item above)
src/base/init/config.c:    if (IS_EMU() && config.speaker==SPKR_NATIVE) { // init, so outside instremu context
src/include/cpu-emu.h:#define IS_EMU_JIT() (IS_EMU() && !config.cpusim) // ok because of !config.cpusim
src/plugin/debugger/mhpdbgc.c:       IS_EMU() ? "emulated," : "", // this could be fine tuned a little (vm86/DPMI) but not super important.
```

Reply:
But what is `if(!IS_EMU() && !(CEmuStat & CeS_INSTREMU))`?
This looks obviously wrong because
!IS_EMU() already means no cpu back-end
uses any emu. So what does the second part
of a condition do?

Reply:
ok, next try, now adjusting EMU_DPMI() and EMU_V86() to take instremu into account.